webhooky/tests/auth.rs

```diff
@@ -0,0 +1,741 @@
+) -> Result<HttpResponseAccepted<String>, HttpError> {
+    Ok(HttpResponseAccepted("ok".to_string()))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/token/verify",
+}]
+async fn token_verification(
+    _rqctx: Arc<RequestContext<()>>,
+    _: QueryToken<InternalToken>,
+) -> Result<HttpResponseAccepted<String>, HttpError> {
+    Ok(HttpResponseAccepted("ok".to_string()))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/token/audit",
+}]
+async fn token_audit(
+    _rqctx: Arc<RequestContext<()>>,
+    _: QueryTokenAudit<InternalToken>,
+) -> Result<HttpResponseAccepted<String>, HttpError> {
+    Ok(HttpResponseAccepted("ok".to_string()))
+}
+
+static INIT: std::sync::Once = std::sync::Once::new();
+
+/// Setup function that is only run once, even if called multiple times.
+fn setup_logger() {
+    INIT.call_once(|| {
+        pretty_env_logger::init();
+    });
+}
+
+fn make_server() -> (u16, HttpServer<()>) {
+    setup_logger();
+
+    // Configure fake test keys for checking implementations
+    std::env::set_var("INTERNAL_AUTH_BEARER", "TEST_BEARER");
+    std::env::set_var("DOCUSIGN_WH_KEY", "vkPkH4G2k8XNC5HWA6QgZd08v37P8KcVZMjaP4zgGWc=");
+    std::env::set_var("GH_WH_KEY", "vkPkH4G2k8XNC5HWA6QgZd08v37P8KcVZMjaP4zgGWc=");
+    std::env::set_var("SLACK_WH_KEY", "vkPkH4G2k8XNC5HWA6QgZd08v37P8KcVZMjaP4zgGWc=");
+
+    let config_dropshot = ConfigDropshot {
+        bind_address: "127.0.0.1:0".parse().unwrap(),
+        request_body_max_bytes: 107374182400, // 100 Gigiabytes.
+        tls: None,
+    };
+    let config_logging = ConfigLogging::StderrTerminal {
+        level: ConfigLoggingLevel::Error,
+    };
+    let log = config_logging.to_logger("webhooky-server").unwrap();
+
+    let mut api = ApiDescription::new();
+    api.register(hmac_github_verification).unwrap();
+    api.register(hmac_github_audit).unwrap();
+    api.register(hmac_checkr_verification).unwrap();
+    api.register(hmac_checkr_audit).unwrap();
+    api.register(hmac_docusign_verification).unwrap();
+    api.register(hmac_docusign_audit).unwrap();
+    api.register(hmac_slack_verification).unwrap();
+    api.register(hmac_slack_audit).unwrap();
+    api.register(hmac_slack_interactive_verification).unwrap();
+    api.register(hmac_slack_interactive_audit).unwrap();
+    api.register(bearer_verification).unwrap();
+    api.register(bearer_audit).unwrap();
+    api.register(token_verification).unwrap();
+    api.register(token_audit).unwrap();
+
+    let api_context = ();
+    let server = HttpServerStarter::new(&config_dropshot, api, api_context.clone(), &log)
+        .map_err(|error| anyhow!("failed to create server: {}", error))
+        .unwrap()
+        .start();
+
+    (server.local_addr().port(), server)
+}
+
+/// General signature tests
+
+#[tokio::test]
+async fn test_missing_signature_hmac_fails() {
+    let (port, _server) = make_server();
+
+    let test_body = include_str!("../tests/github_webhook_sig_test.json").trim();
```

Comment:
Also not a big deal, but isn't the path prefix `../tests` redundant here? I'm also not sure it's worth using external files here, since the bodies all seem quite short. My personal preference would be to use `const` definitions in the source.

Reply:
These are inline now above the tests. Unsure if it is worth using the actual GitHub webhook payload (with certain data redacted), or if GitHub should just use a fake payload like the others.